docs/proposals/proposal-001-trigger-and-deploy.md

```diff
@@ -0,0 +1,248 @@
+they wish to test e.g. `modern-ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
+
+```sh
+curl -X POST \
+     -H "Accept: application/vnd.github.v3+json" \
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"0.2.0", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+The pipeline is versioned by creating releases of the `green-reviews-tooling`
+repo. The git tag to use is passed via the `ref` param.
+
+The CNCF projects will be given a GitHub fine grained access token limited to
+the `green-reviews-tooling` repo. This token will have
+
+- Read access to code and metadata
+- Read write access to actions
+
+### Deploy
+
+Flux is used to deploy the CNCF project. Projects are able to use either
+`kustomization` or `helmrelease` resources to deploy their project.
+
+When the pipeline executes it will look for manifest files in the projects dir.
+If there is a manifest matching the `cncf_project` input its contents will be
+applied using kubectl. The same applies for the `cncf_project_sub` input. 
+
+The `version` param is injected into the files to ensure the correct version of
+the project is deployed. (For these minor changes we can utilize kustomize)
+
+```
+projects
+└── falco
+    ├── ebpf.yaml
+    ├── falco.yaml
+    ├── kmod.yaml
+    └── modern-ebpf.yaml
+```
+
+The pipeline will use a GitHub secret that has a kubeconfig to access the
+green reviews cluster. The deploy step in the pipeline will wait for the newly
+created flux resources to be reconciled before proceeding to the run step.
+
+We will have a node to deploy Falco and another to run the benchmarks
+so we will use [concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)
+to only allow a single execution of the pipeline at any one time.
+
+The separate nodes are a best practice to prevent other components affecting
+the energy measurements.
+
+### Cleanup
+
+On completion of the pipeline run whether it was successful or failed the flux
+resources will be deleted via kubectl. The pipeline will wait for the flux
+resources to be deleted before exiting.
+
+This is to ensure that the cluster state is clean before the next execution of
+the pipeline.
+
+### Graduation Criteria (Optional)
+
+<!--
+List criteria which would allow progression from one maturity level to another.
+eg. What needs to have been accomplished/demonstrated to move from Alpha to Beta.
+
+If applicable, what is the milestone marker which will allow deprecation of the
+replaced capability?
+-->
+
+## Drawbacks (Optional)
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+## Alternatives
```

Comment:
Maybe here as alternative we could link the discussion that we had once with @dipankardas011 about the pull vs push mechanism. From what I got we now decided to go with a pull mechanism - as checking if a new version is deployed - but we also evaluated the push mechanism - the project actively triggering the action, we discarded this and we could maybe document why. What do you think?

Reply:
Push based is still supported. The pipeline can be triggered by calling the GitHub REST API with the token we will provide.

I added pull based after feedback from @leonardpahlke that just push based requires projects to change their CI/CD to add the trigger. 

@ADiTuri @dipankardas011 I added a note on this. WDYT?

Reply:
yes lets add this

and yes why to bother the cncf project maintainers for our pipeline trigger the easy thing would be to have every 30 minutes sync for fetuching latest repo and then trigger. I think this is much more attainable and easier to maintain and secure

Reply:
Yes I also think pull based works better, just wanted to have documented that we had this discussion and we took a decision between the two alternatives. Great like this, feel free to resolve the conversation